### PR TITLE
Enhance `lightdash_project_agent` resource with optional attributes and defaults

### PR DESCRIPTION
**Pull request by yu-iskw**
### **PR Type**
Enhancement, Tests

___

### **Description**
- Make `description`, `space_access`, and `enable_reasoning` optional for `lightdash_project_agent`.

- Set default values for optional `lightdash_project_agent` attributes.

- Remove unused API client and service functions related to evaluations and members.

- Add acceptance tests for minimal `lightdash_project_agent` configuration.

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A[lightdash_project_agent Resource] --> B{Optional Attributes};
  B -- description, space_access, enable_reasoning --> C[Default Values];
  C --> D[Provider Logic];
  D --> E[Lightdash API Client];
  E -- Always send empty spaceAccess --> F[update_agent_v1.go];
  G[Unused API Client/Service Code] --> H[Removed];
  I[Acceptance Tests] --> A;
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Cleanup</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td><strong>append_evaluations_v1.go</strong><dd><code>Remove unused API client for appending evaluations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-8e7b3c32d1d7a1cabeeef453fe181d3727c2fb736b3467f5e63bffe0010e9a75">+0/-71</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>get_evaluation_v1.go</strong><dd><code>Remove unused API client for getting evaluations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-886c34d1e206c41b7990d95d3f0b8be560fe2b09a99e77902d5574d96f73f535">+0/-61</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>get_project_member_by_email_v1.go</strong><dd><code>Remove unused API client for getting project member by email</code></dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-4b421dd30a335486d1105232610c0d8d5f97d8f017650fdc1769a3e343702a2d">+0/-63</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>get_project_member_v1.go</strong><dd><code>Remove unused API client for getting project member by UUID</code></dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-1ab78273e242225691971e8c95e6791db6782f8020025ff44193e3c0e6a7e3d6">+0/-63</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>get_space_v1.go</strong><dd><code>Remove unused `GetSpaceMemberV1` function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-b88385a35b8cc8219cd8e4b6456d93563058f0ba67d5243a5af0c825838874da">+0/-22</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>update_evaluation_v1.go</strong><dd><code>Remove unused API client for updating evaluations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-3b81311a79b8956be16a6e9c91362de8c01c7141929629753f7f3f8eb0eef1e9">+0/-73</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>agent_evaluations.go</strong><dd><code>Remove unused `AppendEvaluations` service function</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-24f690775aa8b76f804907c06522ae9ac356a3b0a646e42e9eee1a8209a06f1e">+0/-20</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td><strong>update_agent_v1.go</strong><dd><code>Ensure `SpaceAccess` is always sent in agent update requests</code></dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-108296db72548892bbb60023911cf7019183bc134bd2b3ed0fa6d4ffbe3dfcf0">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>resource_lightdash_project_agent.go</strong><dd><code>Make agent <code>description</code>, <code>space_access</code>, <code>enable_reasoning</code> optional with <br>defaults</code></dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-a286f90b77ea785a762a2b17e2efc0348ebf1085a37465d8c6b549d8e84a69cb">+29/-18</a>&nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td><strong>update_agent_v1_test.go</strong><dd><code>Add test for `UpdateAgentV1Request` marshaling empty `spaceAccess`</code></dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-d7b1977963f1ebca2281461e0be89c4aed4ea305c6266216693edf99df380357">+40/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>resource_lightdash_project_agent_test.go</strong><dd><code>Add acceptance test for minimal agent configuration with defaults</code></dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-496a336b59e5304ccc2e06cc1a82f2d62d95acaf667c4c5d19b304811b4292fa">+38/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>040_minimal_defaults.tf</strong><dd><code>Add new acceptance test configuration for minimal agent</code>&nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-5ca487d69c9a42e086224f5cbc9ce766e962da40a5c41289b781e9f8712911a0">+21/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Miscellaneous</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>main.go</strong><dd><code>Bump provider version to `0.14.2`</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>resource.tf</strong><dd><code>Add comments for minimal agent configuration example</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/ubie-oss/terraform-provider-lightdash/pull/426/files#diff-1228c68ab0a259aceec90b50a84d5379aa572b0cf32307bf4767c8853e6f21ab">+12/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: internal/provider/resource_lightdash_project_agent.go [hunk 1]
```diff
@@ -742,9 +755,9 @@ func (r *projectAgentResource) Update(ctx context.Context, req resource.UpdateRe
 		userAccess = []string{}
 	}
 
-	// Always send spaceAccess
+	// Space access from plan (empty list means unrestricted access to all spaces)
 	var spaceAccess []string
-	if !plan.SpaceAccess.IsNull() {
+	if !plan.SpaceAccess.IsUnknown() && !plan.SpaceAccess.IsNull() {
 		diags := plan.SpaceAccess.ElementsAs(ctx, &spaceAccess, false)
 		resp.Diagnostics.Append(diags...)
```
[the remark this answers is not in the document]
Comment (yu-iskw) [reply internal/provider/resource_lightdash_project_agent.go:762]: @copilot apply changes based on [this feedback](https://github.com/ubie-oss/terraform-provider-lightdash/pull/426#discussion_r3137768017)
